cli/pkg/config/happy_version_lock.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type HappyVersionLockFile struct {
+	HappyVersion string
+	Path         string `json:"-"`
+}
+
+func NewHappyVersionLockFile(projectRoot string, requiredVersion string) (*HappyVersionLockFile, error) {
+
+	if projectRoot == "" {
+		return nil, errors.New("No projectRoot specified")
+	}
+
+	if requiredVersion == "" {
+		return nil, errors.New("No requiredVersion specified")
+	}
+
+	path := calcHappyVersionPath(projectRoot)
+
+	return &HappyVersionLockFile{
+		HappyVersion: requiredVersion,
+		Path:         path,
+	}, nil
+}
+
+func DoesHappyVersionLockFileExist(projectRoot string) bool {
+	filePath := calcHappyVersionPath(projectRoot)
+	_, err := os.Stat(filePath)
```

Comment:
I think os.Stat errors can be varied. You might need to do something like this:

~~~
_, err := os.Stat("/file/that/exists");

if err != nil {
	if os.IsNotExist(err) {
		// file does not exist, do something
	} else {
		// more serious errors
	}
}
~~~

Reply:
done